### ISSUE TITLE
Steam closing completely on stream disconnect

### ISSUE THREAD

**Issue by CommonMugger** [ClassicOldSong/Apollo#138]
This seems to be a problem with Sunshine as well. If, for example, I'm on my Steam Deck and I put the stream to sleep by hitting the power button, Steam will completely crash and I will need to relaunch the app I was using. 

This happens with any hardware I'm using. So far, I've tested: 

Google TV
Apple TV
iPad
iPhone
I don't exit the app, just disconnect the Stream. I usually play single player games so I'm hoping to just be able to pause the game and disconnect the stream, then resume it to have my game waiting rather than restart the game completely.

Any idea what's causing this?

**Comment by ClassicOldSong** [comment]
What's your host's configuration? Windows version, hardware spec etc.

Also disable Steam Record, it might cause conflicts with video encoders.

**Comment by CommonMugger** [comment]
Windows 11 23H2
64gb of DDR 4 ram 3200
RTX 3080

Steam record seems to have been off this entire time, never enabled it.

**Comment by ClassicOldSong** [comment]
Hmmmm never got this issue. All games/applications or only games launched through Steam have this issue?

**Comment by CommonMugger** [comment]
I haven't tested a non Steam game. I use shortcuts I set up through my script (can find it in my public repo) but it doesn't do anything weird. I'll test when I get home.

**Comment by CommonMugger** [comment]
I tested it - yeah it still seems to happen. I'm not sure what the cause is.

**Comment by ClassicOldSong** [comment]
Maybe you need to make a clean WindowsToGo disk and test again there

I suspect it's an issue with your graphics card driver. The only thing happens when a stream disconnects is that the encoder releases the resources it occupies and then stops encoding, normally it shouldn't affect the running program.

**Comment by CommonMugger** [comment]
Yeah I was actually planning on just wiping my entire PC anyways and starting from scratch, so this is just a good excuse to do it sooner. I'll update with what happens.

**Comment by ClassicOldSong** [comment]
Any updates?

**Comment by CommonMugger** [comment]
Yes actually, sorry. It's still happening after a format. I think it has something to do with the Apollo app shortcuts. I'm using the appid in order to launch the games. I need to test with just the EXEs instead and see if that slant crash steam, I just haven't gotten around to it yet.

**Comment by ClassicOldSong** [comment]
That's really strange...

**Comment by CommonMugger** [comment]
I got a Logitech G Cloud and started using Artemis, and it's no longer happening. I guess we can assume that the Steam Deck version of Moonlight is the problem then?

**Comment by ClassicOldSong** [comment]
SteamDeck version is just the desktop version. Did you have "Quit app on host PC after ending stream" checked in the client settings accidentally?

**Comment by CommonMugger** [comment]
Just checked, nope. Just to rule out issues completely with the Deck, I'm gonna factory reset it and re-setup moonlight on it, try again.

**Comment by CommonMugger** [comment]
Not sure why it's still happening on the deck but even after factory reset, it is. I'll just close this since I use a different device more often.
